src/js/configuration_panel/panel-events.js

```diff
@@ -59,7 +66,7 @@ https://github.com/fluid-lab/gamepad-navigator/blob/master/LICENSE
          */
         fluid.each(inputMenusArray, function (inputMenu) {
             if (fluid.isDOMNode(inputMenu)) {
-                that.changeConfigMenuOptions(inputMenu.querySelector("select"));
+                that.changeConfigMenuOptions(inputMenu.querySelector(".action-dropdown"));
```

Comment:
I can see that you're kind of halfway using the viewComponent infrastructure.  When you get around to working on #40, you'll probably need to rework this to either render all the controls and show/hide them as needed, or otherwise avoid reusing a single set of controls for all inputs.  You'll also need to migrate to using `that.locate` rather than directly querying the DOM using jQuery.

If you could leave a TODO on the function that backs the `changeConfigMenuOptions` invoker, that'd be enough for now.

Reply:
> you'll probably need to rework this to either render all the controls and show/hide them as needed, or otherwise avoid reusing a single set of controls for all inputs.

@the-t-in-rtf thanks for the feedback, but I'm slightly confused with this statement. The approach we are currently using does the same, i.e., render all the controls and display/hide them according to the action selected by the user. Also, I'm not sure what you mean by the second part of the statement. Could you please elaborate on this a bit?

Reply:
I had mistakenly thought you were reusing a single element because of your strategy of determining which select it was using the classname.   I guess that instead you're using a general method for all selects and determining which you're working with using the classname.

In any case, my point is still that it would be better to make a sub-component for select elements in general, and to have each sub-component manage the mapping for one control and manage its visibility and settings.  The settings would be relayed into the parent component.